Source/Core/Core/NetPlayClient.cpp

```diff
@@ -248,7 +248,12 @@ bool NetPlayClient::Connect()
   sf::Packet rpac;
   // TODO: make this not hang
   ENetEvent netEvent;
-  if (enet_host_service(m_client, &netEvent, 5000) > 0 && netEvent.type == ENET_EVENT_TYPE_RECEIVE)
+  int net;
+  while ((net = enet_host_service(m_client, &netEvent, 5000)) > 0 && netEvent.type == 42)
```

Comment:
I have no familiarity with this part of Dolphin, but it looks like the number 42 isn't defined [in `ENetEventType`](https://github.com/dolphin-emu/dolphin/blob/master/Externals/enet/include/enet/enet.h#L394-L423) but instead inserted [in Dolphin's `ENetUtil::InterceptCallback`](https://github.com/dolphin-emu/dolphin/blob/f4b1155b4d1d0b6ef49236eb3adfd3437304a7c1/Source/Core/Common/ENetUtil.cpp#L29-L38). I'm not sure why it works like this, or if it'd be feasible to just get rid of that code, though.

It also seems like this could result in an infinite loop, but that may have been the case before given the "TODO: make this not hang" above.

Reply:
I suspect the 42 is inserted so that the traversal server packets get ignored by the rest of the netplay code.

This could definitely result in an infinite loop though, now you mention it: pings are probably about once a second, and the timeout here is 5 seconds. The easy fix would be to have a local timer that's incremented as server packets are received, and wait for that to cross the 5 second mark.

Reply:
Actually come to think of it, this blocks the thread, so the client wouldn't send any pings to the server, even if it took the full 5 seconds.

Reply:
Just to make sure, I checked what happens if a valid response is never sent, and yeah it times out properly.